### PR TITLE
Added ethon easy specs

### PR DESCRIPTION
**Pull request by MirkoMignini**

### CODE REVIEW

File: lib/sharepoint/client.rb [hunk 1]
```diff
@@ -346,8 +346,12 @@ def ethon_easy_json_requester
       easy
     end
 
+    def ethon_easy_options
+      config.ethon_easy_options ||= {}
```
Comment (lleirborras) [inline lib/sharepoint/client.rb]: why `||=` ? Do not alter `config` just memoize or `||`

File: lib/sharepoint/client.rb [hunk 2]
```diff
@@ -346,8 +346,12 @@ def ethon_easy_json_requester
...
+    end
+
     def ethon_easy_requester
-      easy = Ethon::Easy.new(httpauth: :ntlm, followlocation: 1, maxredirs: 5)
+      easy = Ethon::Easy.new({httpauth: :ntlm, followlocation: 1, maxredirs: 5}.merge(ethon_easy_options))
```
Comment (lleirborras) [inline lib/sharepoint/client.rb]: Are we sure we do not wanna `reverse_merge` ?

**Comment by algodave** [comment]
@MirkoMignini We might want to validate the new option in the `validate_config!` method, I would suggest to check that it is a `Hash`

**Comment by algodave** [comment]
👍 for me

**Pull request merged**